starboard/build/config/starboard_target_type.gni

```diff
@@ -23,3 +23,30 @@ if (starboard_target_type == "") {
     starboard_target_type = "group"
   }
 }
+
+template("starboard_platform_target") {
+  target(starboard_target_type, target_name) {
+    forward_variables_from(invoker, [ "ldflags" ])
```

Comment:
Instead of forwarding ldflags, put the cast ldflags in a config and add it to a list `extra_configs` which we forward here, then add to existing configs. See examples here: https://source.chromium.org/search?q=case:y%20%5CWextra_configs%5CW&sq=&ss=chromium%2Fchromium%2Fsrc